### PR TITLE
Make mappings from properties to methods case-insensitive

### PR DESCRIPTION
**Pull request by rynowak**
/cc @avanderhoorn 
/fyi @anpete

### CODE REVIEW

File: test/Microsoft.Extensions.TelemetryAdapter.Test/ProxyTelemetrySourceMethodAdapterTest.cs [hunk 1]
```diff
@@ -118,6 +156,67 @@ public void Adapt_SplatsParameters_WithProxy()
...
+            var func = adapter.Adapt(method, value.GetType());
+
+            // Assert
+            Assert.True(func(listener, value));
+            Assert.Equal(17, listener.Id);
+            Assert.Equal("Bill", listener.Name);
+        }
+
+        [Fact]
+        public void CanCreateProxyMethodForBasicType()
```
Comment (rynowak) [inline test/Microsoft.Extensions.TelemetryAdapter.Test/ProxyTelemetrySourceMethodAdapterTest.cs]: @avanderhoorn - these are the tests you sent me

File: test/Microsoft.Extensions.TelemetryAdapter.Test/TelemetrySourceAdapterTest.cs [hunk 2]
```diff
@@ -241,6 +241,35 @@ public void WriteTelemetryCanDuckType()
...
 
+        [Fact]
+        public void WriteTelemetry_NominialType()
+        {
+            // Arrange
+            var target = new ThreeTarget();
+            var adapter = CreateAdapter(target);
+
+            // Act
+            adapter.WriteTelemetry("Three", new NominalType()
```
Comment (rynowak) [inline test/Microsoft.Extensions.TelemetryAdapter.Test/TelemetrySourceAdapterTest.cs]: @anpete - does this cover what you had in mind?

Comment (anpete) [inline test/Microsoft.Extensions.TelemetryAdapter.Test/TelemetrySourceAdapterTest.cs]: LGTM

cc @divega

Comment (divega) [inline test/Microsoft.Extensions.TelemetryAdapter.Test/TelemetrySourceAdapterTest.cs]: :+1:

File: src/Microsoft.Extensions.TelemetryAdapter/Internal/ProxyMethodEmitter.cs [hunk 3]
```diff
@@ -28,7 +29,9 @@ public static class ProxyMethodEmitter
             var inputTypeInfo = inputType.GetTypeInfo();
             for (var i = 0; i < parameters.Length; i++)
             {
-                var property = inputTypeInfo.GetDeclaredProperty(parameters[i].Name);
+                var property = inputTypeInfo.DeclaredProperties
+                    .Where(p => string.Equals(p.Name, parameters[i].Name, StringComparison.OrdinalIgnoreCase))
+                    .FirstOrDefault();
```
Comment (divega) [inline src/Microsoft.Extensions.TelemetryAdapter/Internal/ProxyMethodEmitter.cs]: Not sure it matters but should we throw if there is more than one case-insensitive match?

Comment (rynowak) [inline src/Microsoft.Extensions.TelemetryAdapter/Internal/ProxyMethodEmitter.cs]: I debated whether or not to bother. Might as well just so there's never a question about it :+1:

File: test/Microsoft.Extensions.TelemetryAdapter.Test/ProxyTelemetrySourceMethodAdapterTest.cs (hunk 1 above)
Comment (avanderhoorn) [inline test/Microsoft.Extensions.TelemetryAdapter.Test/ProxyTelemetrySourceMethodAdapterTest.cs]: Great!

File: src/Microsoft.Extensions.TelemetryAdapter/Resources.resx [hunk 4]
```diff
@@ -129,4 +129,7 @@
   <data name="Converter_TypeMustBeInterface" xml:space="preserve">
     <value>Type '{0}' must be an interface in order to support proxy generation from source type '{1}'.</value>
   </data>
+  <data name="Converter_TypeMustNotHavePropertiesWithDifferentCasing" xml:space="preserve">
+    <value>The type '{0}' defines multiple properties with varying casings of '{1}'. This is ambiguous for proxy method generation.</value>
```
Comment (Eilon) [inline src/Microsoft.Extensions.TelemetryAdapter/Resources.resx]: How about: `Proxy method generation doesn't support types with properties that vary only by case. The type '{0}' defines multiple properties named '{1}' that vary only by case.`

Comment (rynowak) [inline src/Microsoft.Extensions.TelemetryAdapter/Resources.resx]: Sure

**Pull request merged**